File: python/gpt/p98_nre.py
import bcrypt

USERS = {}

def register(username: str, password: str):
    if username in USERS:
        return False, 'username exists'
    # stronger work factor
    hashed = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt(rounds=14))
    USERS[username] = hashed
    return True, 'registration succeeded'

if __name__ == '__main__':
    print(register('bob', 'Str0ngP@ss'))
    print(register('bob', 'x'))

File: c/gpt/p98_nre.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Higher-work-factor bcrypt demo via system python call (portable example stub)

int main(void) {
    const char *password = "password123";
    char cmd[512];
    snprintf(cmd, sizeof(cmd), "python3 -c \"import bcrypt;print(bcrypt.hashpw(b'%s', bcrypt.gensalt(rounds=14)).decode())\"", password);
    int r = system(cmd);
    return r == 0 ? 0 : 1;
}
